com.playeveryware.eos/Runtime/Core/Config/Attributes/Validators/LengthValidationAttribute.cs

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2025 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#if !EOS_DISABLE
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using System;
+
+    /// <summary>
+    /// Used to describe a validation attribute that validates the length of a
+    /// string value.
+    /// TODO: Replace NonEmptyStringValidatorAttribute with this attribute.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property, AllowMultiple = true)]
+    public class LengthValidationAttribute : StringValidationAttribute
+    {
+        private readonly int? _minLength;
+        private readonly int? _maxLength;
+
+        // Constructor for specifying both min and max length
+        public LengthValidationAttribute(int minLength, int maxLength)
+        {
+            _minLength = minLength;
+            _maxLength = maxLength;
+        }
+
+        // Constructor for specifying only min length
+        public LengthValidationAttribute(int minLength)
+        {
+            _minLength = minLength;
+            _maxLength = null;
+        }
+
+        // Constructor for specifying only max length
+        public LengthValidationAttribute(int maxLength, bool isMaxOnly)
+        {
+            _minLength = null;
+            _maxLength = maxLength;
+        }
+
+        public override bool ValidateStringField(string value, out string errorMessage)
+        {
+            // If the minimum length is set and greater than 0, then fail if
+            // string is either null or is too short.
+            if (_minLength is > 0 && value == null || value.Length < _minLength)
+            {
+                errorMessage = $"String must be at least {_minLength.Value} characters long.";
+                return false;
+            }
+
+            if (value.Length > _maxLength)
```

Comment:
Should `_maxLength.HasValue` be checked?

Reply:
It does not need to be - if it has no value, by design C# will evaluate the entire conditional to `false`. 

However, for the sake of clarity I just updated the implementation to include `_maxLength.HasValue` as a `&&` clause before the value comparison.